clang/lib/CodeGen/CGCall.cpp

```diff
@@ -5894,7 +5894,22 @@ RValue CodeGenFunction::EmitCall(const CGFunctionInfo &CallInfo,
       bool isFp32SqrtFunction =
           (FuncName == "sqrt" && !getLangOpts().OffloadFP32PrecSqrt &&
            IsFloat32Type);
-      if (hasFPAccuracyFuncMap || hasFPAccuracyVal || isFp32SqrtFunction) {
+      bool ArgsTypeIsFloat = true;
+      // In sycl mode, functions' arguments of type half are expanded
+      // to pointer types. Exclude these functions from bein emitted
+      // as fpbuiltins.
+      if (!getLangOpts().OffloadFP32PrecSqrt ||
```

Comment:
Do you mean **and** instead?
```suggestion
      if (!getLangOpts().OffloadFP32PrecSqrt &&
```

Reply:
@premanandrao I think we want an `||` here. We want to restrict if any of `-fno-offload-fp32-prec-sqrt` or `-fno-offload-fp32-prec-div` is used.
We could have things like `a / sqrt(half p)` or sqrt (half p1 / half p2) ... .